packages/expo-cli/src/commands/eas-build/build/metadata.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  AndroidBuildProfile,
+  CredentialsSource,
+  Workflow,
+  iOSBuildProfile,
+} from '../../../easJson';
+import { CommandContext } from './types';
+
+/**
+ * We use require() to exclude package.json from TypeScript's analysis since it lives outside
+ * the src directory and would change the directory structure of the emitted files
+ * under the build directory
+ */
+const packageJSON = require('../../../../package.json');
+
+export type BuildMetadata = {
+  /**
+   * Application version (the expo.version key in app.json/app.config.js)
+   */
+  appVersion: string;
+
+  /**
+   * Expo CLI version
+   */
+  cliVersion: string;
+
+  /**
+   * Build workflow
+   * It's either 'generic' or 'managed'
+   */
+  workflow: Workflow;
+
+  /**
+   * Credentials source
+   * Credentials could be obtained either from credential.json or Expo servers.
+   * Auto mode means that both sources will be checked.
```

Comment:
I think that auto should not be passed here this value should specify what credentials eventually were used